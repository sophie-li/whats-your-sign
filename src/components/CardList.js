import React from 'react';
import Card from './Card';

const CardList = ({ signs }) => {
  console.log('SIGNS ---', signs);
  return (
    <div className='flex flex-wrap justify-center'>
      {signs.map((sign, index) => {
        const {
          id,
          name,
          spiritColor,
          luckyNumbers,
          symbol,
          luckyDay,
          rulingPlanet,
          qualities,
          emojis,
        } = signs[index];
        return (
          <Card
            key={index}
            id={id}
            name={name}
            symbol={symbol}
            spiritColor={spiritColor}
            luckyNumbers={luckyNumbers}
            luckyDay={luckyDay}
            rulingPlanet={rulingPlanet}
            qualities={qualities}
            emojis={emojis}
          />
        );
      })}
    </div>
  );
};

export default CardList;
