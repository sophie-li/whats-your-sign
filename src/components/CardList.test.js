import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('expect to render CardList component', () => {
  const mockSigns = [
    {
      id: 1,
      startDate: '03-21',
      endDate: '04-19',
      name: 'aries',
      sign: 'fire',
      emoji: 'ram',
      symbol: 'the ram',
      spiritColor: 'red',
      luckyNumbers: [1, 9],
      luckyDay: 'tuesday',
      rulingPlanet: 'Mars',
      qualities: ['brave', 'optimistic', 'passionate', 'active', 'energetic'],
      emojis: {
        name: '♈',
        symbol: '🐏',
        planet: '♂️',
      },
    },
  ];

  expect(shallow(<CardList signs={mockSigns} />)).toMatchSnapshot();
});
