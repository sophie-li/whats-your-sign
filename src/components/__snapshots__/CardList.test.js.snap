// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<div
  className="flex flex-wrap justify-center"
>
  <Card
    emojis={
      Object {
        "name": "♈",
        "planet": "♂️",
        "symbol": "🐏",
      }
    }
    id={1}
    key="0"
    luckyDay="tuesday"
    luckyNumbers={
      Array [
        1,
        9,
      ]
    }
    name="aries"
    qualities={
      Array [
        "brave",
        "optimistic",
        "passionate",
        "active",
        "energetic",
      ]
    }
    rulingPlanet="Mars"
    spiritColor="red"
    symbol="the ram"
  />
</div>
`;
